pdp/assignment.go

```diff
@@ -460,6 +469,20 @@ func (a AttributeAssignment) String() string {
 	return fmt.Sprintf("(%s)%s:%s", valueType, name, value)
 }
 
+// MarshalJSON satisfies Marshaler interface
+func (a AttributeAssignment) MarshalJSON() ([]byte, error) {
+	name, valueType, value, err := a.Serialize(nil)
```

Comment:
Generally, `Serialize` panics in its argument is `nil`. It is a consequence of attempt to represent any expression as value - generally, it isn't possible. For now you can add check that `a.e` is instance of `AttributeValue`. In that case serialization works with no context. For other type you can put something like `"<complicated expression>"` instead of value. Later you need to add `MarshalJSON` to `Expression` interface and to implement it for all structures which implementing `Expression` interface. There are bunch of them starting with `AttributeValue`, `AttributeDesignator`, and so on. Actually, there are so many of them that implementation of marshaling for them is main part of this work.

Reply:
I think it would also be useful, e.g. for diagnostics purposes, to be able to Serialize (at least not panic) even when the Expression cannot be evaluated due to missing `context` - in which case, we could just include whatever information that is available.